Saturation/SaturationAlgorithm.cpp

```diff
@@ -541,6 +574,7 @@ void SaturationAlgorithm::onClauseReduction(Clause* cl, Clause* replacement,
   }
 
   if (replacement) {
+    //TODO fix this
     onParenthood(replacement, cl);
```

Comment:
Is that thing that should be fixed the fact, that only the first replacement will have "onParenthood" called?

Did you decide it's OK never to change this?

Reply:
That is correct. Assuming we never run symbol elimination along with higher-order we should be safe (looking at onParenthood). I have made the comment more explanatory.

Reply:
Thanks!